pkg/controllers/networkpolicy/namespace_handler.go

```diff
@@ -0,0 +1,97 @@
+/* Copyright © 2025 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package networkpolicy
+
+import (
+	"context"
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/util"
+)
+
+// EnqueueRequestForNamespace handles Namespace events and triggers NetworkPolicy reconciliation
+// when Namespace labels change and there are pods in the namespace.
+type EnqueueRequestForNamespace struct {
+	Client                  client.Client
+	NetworkPolicyReconciler *NetworkPolicyReconciler
+}
+
+func (e *EnqueueRequestForNamespace) Create(_ context.Context, _ event.CreateEvent, _ workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	log.V(1).Info("Namespace create event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Delete(_ context.Context, _ event.DeleteEvent, _ workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	log.V(1).Info("Namespace delete event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Generic(_ context.Context, _ event.GenericEvent, _ workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	log.V(1).Info("Namespace generic event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Update(_ context.Context, updateEvent event.UpdateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	obj := updateEvent.ObjectNew.(*v1.Namespace)
+	if isInSysNs, err := util.IsSystemNamespace(nil, "", obj); err != nil {
```

Comment:
For VPC mode, we cannot use `IsSystemNamespace` to check whether it is system namespace, right?

Reply:
fixed

Reply:
I think here you can use e.NetworkPolicyReconciler.Service.NSXConfig.EnableVPCNetwork to check if VPC mode is enabled or not.